openqa-qam-sle

```diff
@@ -0,0 +1,220 @@
+    ID="$(echo "$ReviewRequestID" | awk -F':' '{printf "%s", $3}')" # The ID of the update
+    # local RR="$(echo "$ReviewRequestID" | awk -F':' '{printf "%s", $4}')" # The Release Request of the update
+
+    local incident_settings
+    incident_settings=$(curl -sk -X GET "http://dashboard.qam.suse.de/api/incident_settings/$ID")
+
+    local build
+    build=$(echo "$incident_settings" | jq -r '.[0].settings.BUILD')
+
+    local versions
+    versions=$(echo "$incident_settings" | jq -r '.[].settings.VERSION' | sort | uniq)
+
+    local versions_teradata
+    versions_teradata=$(echo "$incident_settings" | jq -r '[.[]| { KEY:(.settings.VERSION + "|" +.settings.FLAVOR)}] | .[].KEY' | sort | uniq | grep TERADATA | sed -e 's/|.*-TERADATA$/-TERADATA/' | sort | uniq)
+
+    for version in $versions $versions_teradata; do
+        # echo "# Version: $version"
+        local groupid=${dict_group[$version]}
+        [[ -z "$groupid" ]] && continue
+
+        local count_url="https://openqa.suse.de/api/v1/jobs/overview?distri=sle&version=${version%%-TERADATA}&build=${build}&groupid=${groupid}"
+        local count
+        count=$(curl -sk -X GET "$count_url" | jq -r '. | length')
+        [[ $count -eq 0 ]] && continue
+
+        printf 'Version: "%s" Build: "%s"\n' "$version" "$build"
+        version=${version%%-TERADATA}
+        printf 'https://openqa.suse.de/tests/overview?distri=sle&version=%s&build=%s&groupid=%s\n' "$version" "$build" "$groupid"
+
+        local running_url="https://openqa.suse.de/api/v1/jobs/overview?distri=sle&version=${version}&build=${build}&groupid=${groupid}&state=scheduled&state=running"
+        local running
+        running=$(curl -sk -X GET "$running_url" | jq -r '. | length')
+        [[ $running -gt 0 ]] && {
+            _warn "RUNNING / SCHEDULED ($running jobs) Awaiting completion..."
+            echo ""
+            continue
+        }
+
+        local failed_url="https://openqa.suse.de/api/v1/jobs/overview?distri=sle&version=${version}&build=${build}&groupid=${groupid}&result=failed&result=incomplete&result=timeout_exceeded"
+        local failed
+        failed=$(curl -sk -X GET "$failed_url" | jq -r '. | length')
+        [[ $failed -eq 0 ]] && _ok "PASSED"
+        [[ $failed -gt 0 ]] && _ko "FAILED ($failed jobs)"
+        echo ""
+    done
+    echo "---"
+}
+
+search_maintenance_aggregated() {
+    _title "Maintenance: Aggregated updates / Core Maintenance Updates"
+    local ReviewRequestID=$1
+    local days=${2:-5}  # Default last 5 days
+    local groupid="414" # Maintenance: Aggregated updates / Core Maintenance Updates
+
+    local ID
+    ID="$(echo "$ReviewRequestID" | awk -F':' '{printf "%s", $3}')" # The ID of the update
+    # local RR="$(echo "$ReviewRequestID" | awk -F':' '{printf "%s", $4}')" # The Release Request of the update
+
+    local versions
+    versions=$(curl -sk -X GET "http://dashboard.qam.suse.de/api/incident_settings/$ID" | jq -r '.[].settings.VERSION' | sort | uniq)
```

Comment:
Yes I agree, but the original idea was to have an independent function to reuse with no global variables to depend on. This also explains why I preferred passing parameters to functions rather than using global variables

Reply:
I think for something like this it is totally fine to use a global variable.

Reply:
Replaced hard-coded url with readonly var